src/hiprtc.cpp

```diff
@@ -354,27 +387,25 @@ namespace
 {
     class Unique_temporary_path {
         // DATA
-        std::experimental::filesystem::path path_{};
+        std::string path_{};
     public:
         // CREATORS
         Unique_temporary_path() : path_{std::tmpnam(nullptr)}
         {
-            while (std::experimental::filesystem::exists(path_)) {
+            while (hip_impl::exists(path_)) {
                 path_ = std::tmpnam(nullptr);
             }
         }
-        Unique_temporary_path(const std::string& extension)
-            : Unique_temporary_path{}
-        {
-            path_.replace_extension(extension);
-        }
 
         Unique_temporary_path(const Unique_temporary_path&) = default;
         Unique_temporary_path(Unique_temporary_path&&) = default;
 
         ~Unique_temporary_path() noexcept
         {
-            std::experimental::filesystem::remove_all(path_);
+            std::string s("rm -r " + path_);
+            if (path_.substr(0, 5) == "/tmp/") {
+                system(s.c_str());
```

Comment:
Is there a better way to remove the tmp directory other than spawn a new process?